google-cloud-firestore/lib/google/cloud/firestore/client.rb

```diff
@@ -416,17 +416,13 @@ def transaction max_retries: nil
         ##
         # @private
         def coalesce_get_argument obj
-          if obj.is_a?(String) || obj.is_a?(Symbol)
-            if obj.to_s.split("/").count.even?
-              return doc obj # Convert a DocumentReference
-            else
-              return col obj # Convert to Query
-            end
-          end
-
           return obj.ref if obj.is_a? DocumentSnapshot
```

Comment:
I don't see this logic/behavior in the old version. Is this more than just a refactoring?